### PR TITLE
Update code to ES6

### LINKED ISSUE #416: Use ES6 modules
We're currently using the non-standard `require`. We should instead migrate to `import`.

### PR DESCRIPTION
**Pull request by Sarthak-Dayal**
Update some code to ES6 to address (fixes #416).

Specifically,

- [X] Replaced all require statements with imports
- [X] Changed var to let keywords

This helps ensure that all of the code is up to date with ES6 standards and allows for easier code maintenance.

Checklist for reviewer:

- [x] The description should reference a bug or github issue, if relevant.
- [ ] There must be a [`CHANGELOG.md`](./CHANGELOG.md) entry for any non-test change.
- [ ] Any change to the NPM commands must be carefully reviewed to make sure it won't break the Add-ons pipeline.
- [ ] Any version increase must follow the [release process](./docs/RELEASE_PROCESS.md).

### CODE REVIEW

File: core-addon/background.js [hunk 1]
```diff
@@ -2,8 +2,9 @@
  * License, v. 2.0. If a copy of the MPL was not distributed with this
  * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
 
-window.browser = require("webextension-polyfill");
-const Core = require("./Core.js");
+import { Core } from "./Core.js"
+import * as browser from "webextension-polyfill";
+window.browser = browser;
```
Comment (Dexterp37) [inline core-addon/background.js:7]:
```suggestion
import "webextension-polyfill";
```

File: core-addon/content-script.js [hunk 2]
```diff
@@ -2,7 +2,8 @@
  * License, v. 2.0. If a copy of the MPL was not distributed with this
  * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
 
-window.browser = require("webextension-polyfill");
+import * as browser from "webextension-polyfill";
+window.browser = browser;
```
Comment (Dexterp37) [inline core-addon/content-script.js:6]:
```suggestion
import "webextension-polyfill";
```

**Review by Dexterp37 — CHANGES_REQUESTED** [review]
Thank you for your contribution, @Sarthak-Dayal!

This looks good overall, there's a few minor changes requested (see my comments), and two main things that we need to change in order to make tests work.

1. Due to how ES6 `import` support works in MochaJS, we need to rename our unit test files to have the `mjs` extension instead of just `js`. This should make the test errors detected on CI disappear.
2. The command to run [unit tests](https://github.com/mozilla-rally/rally-core-addon/blob/4ddab3db11663d8cd80cd9bf1e0821fadc1764df/package.json#L17) must be changed to look for `mjs` files

**Comment by Sarthak-Dayal** [comment]
Hey @Dexterp37! I will work on the required changes right now. Thank you for your review.

**Comment by Sarthak-Dayal** [comment]
I fixed a few issues (will push those changes soon), but one of the main problems I am having is with `hooks.mjs` not recognizing `sinon-chrome/extensions` as a valid import. @Dexterp37 Would you be able to help me with this?

File: core-addon/background.js [hunk 3]
```diff
@@ -1,13 +1,17 @@
-/* This Source Code Form is subject to the terms of the Mozilla Public
+/**
```
Comment (Dexterp37) [inline core-addon/background.js:1]: Would you kindly revert the changes to the copyright notice here? :-)

File: core-addon/background.js [hunk 4]
```diff
@@ -1,13 +1,17 @@
...
+ */
 
-window.browser = require("webextension-polyfill");
-const Core = require("./Core.js");
+import "./Core.js";
+import "webextension-polyfill";
 
 const core = new Core({
-    availableStudiesURI: __STUDIES_LIST__,
-    disableRemoteSettings: __DISABLE_REMOTE_SETTINGS__,
```
Comment (Dexterp37) [inline core-addon/background.js:10]: Would you kindly revert these changes? :)

File: tests/hooks.mjs [hunk 5]
```diff
@@ -1,15 +1,19 @@
-/* This Source Code Form is subject to the terms of the Mozilla Public
+/**
+ * /* This Source Code Form is subject to the terms of the Mozilla Public
  * License, v. 2.0. If a copy of the MPL was not distributed with this
- * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ *
+ * @format
+ */
```
Comment (Dexterp37) [inline tests/hooks.mjs:7]: Would you kindly revert these changes?

File: tests/hooks.mjs [hunk 6]
```diff
@@ -1,15 +1,19 @@
-/* This Source Code Form is subject to the terms of the Mozilla Public
```
Comment (Dexterp37) [inline tests/hooks.mjs:1]: I don't think we strictly need to rename this file. We can leave it untouched.

File: package.json [hunk 7]
```diff
@@ -28,6 +28,7 @@
     "@storybook/addon-essentials": "^6.1.18",
     "@storybook/addon-links": "^6.1.18",
     "@storybook/svelte": "^6.1.18",
+    "@types/sinon-chrome": "^2.2.10",
```
Comment (Dexterp37) [inline package.json:31]: We don't seem to be using this dependency. Let's kindly drop it.

File: package.json [hunk 8]
```diff
@@ -39,7 +40,7 @@
     "eslint-plugin-svelte3": "^3.1.1",
     "geckodriver": "^1.22.1",
     "micromodal": "^0.4.6",
-    "mocha": "^8.2.1",
+    "mocha": "^8.3.0",
```
Comment (Dexterp37) [inline package.json:43]: Let's not bump versions in this PR, please :)

**Review by Dexterp37 — CHANGES_REQUESTED** [review]
Thanks again for your contribution! I left a few comments below. In addition to these, note that we need to export classes for them to be importable. This means, for example, that we need to replace `module.exports = class ...` with `export default class ...` in our code, otherwise compilation will break.

You can test this by reverting the renaming of `hooks.mjs` and then issuing `npm run test-addon`.

**Comment by Sarthak-Dayal** [comment]
Changing `hooks.mjs` to `hooks.js` causes the typical mocha error of "Cannot use import statement outside a module" because sinon-chrome and webextension-polyfill are being imported. The only way I see to solve this is to change it back to `.mjs` Is there anything else you had in mind? @Dexterp37

**Comment by Dexterp37** [comment]
> 
> 
> Changing `hooks.mjs` to `hooks.js` causes the typical mocha error of "Cannot use import statement outside a module" because sinon-chrome and webextension-polyfill are being imported. The only way I see to solve this is to change it back to `.mjs` Is there anything else you had in mind? @Dexterp37

You can leave the the `hooks.js` file unmodified, without migrating it to ES6 modules for now.

**Comment by Dexterp37** [comment]
Hey @Sarthak-Dayal ! Are you still considering working on this? :)

**Comment by Sarthak-Dayal** [comment]
Hey @Dexterp37 ! Yes, I apologise that i wasn't able to contribute much. I got really busy with other things. I will certainly try to get the PR completely ready this week.

**Comment by Dexterp37** [comment]
> 
> 
> Hey @Dexterp37 ! Yes, I apologise that i wasn't able to contribute much. I got really busy with other things. I will certainly try to get the PR completely ready this week.

No need to apoligise, don't worry! No pressure, I just wanted to make sure you were still interested in doing this :-)

**Comment by Dexterp37** [comment]
@Sarthak-Dayal apologies :-( This was bumped in priority due to a dependency and we need to work on this, so I'm finishing it up. If you are still interested in contributing, would you consider [this bug](https://bugzilla.mozilla.org/show_bug.cgi?id=1695884)?

**Comment by Dexterp37** [comment]
Closed in favour of #507

**Pull request closed without merging**